src/main/kotlin/no/nav/syfo/api/endpoints/VedtakEndpoints.kt

```diff
@@ -57,11 +60,14 @@ fun Route.registerVedtakEndpoints(
 
             val personident = call.getPersonident()
                 ?: throw IllegalArgumentException("Failed to $API_ACTION: No $NAV_PERSONIDENT_HEADER supplied in request header")
+            val token = call.getBearerHeader() ?: throw IllegalArgumentException("Failed to $API_ACTION: No bearer token supplied in request header")
             val navIdent = call.getNAVIdent()
             val callId = call.getCallId()
 
             if (vedtakService.getVedtak(personident).any { !it.isFerdigbehandlet() }) {
                 call.respond(HttpStatusCode.Conflict, "Finnes allerede et åpent vedtak for personen")
+            } else if (!arbeidssokeroppslagClient.isArbeidssoker(callId, personident, token)) {
+                call.respond(HttpStatusCode.BadRequest, "Personen er ikke registrert som arbeidssøker")
```

Comment:
Burde dette kallet ligget som en sjekk i `vedtakService.createVedtak` i stedet for direkte her? Målet er kanskje å kunne gi en egen responskode for denne?

Reply:
Ja, forsøkte det først, men synes denne løsningen ble ryddigere.